src/mapgen/mapgen_v7.cpp

```diff
@@ -446,6 +486,36 @@ int MapgenV7::generateTerrain()
 		noise_mountain->perlinMap3D(node_min.X, node_min.Y - 1, node_min.Z);
 	}
 
+	//// Floatlands
+	// 'Generate floatlands in this mapchunk' bool for
+	// simplification of condition checks in y-loop.
+	bool gen_floatlands = false;
+	float *float_offset_cache = new float[csize.Y + 2];
```

Comment:
Please move this allocation to the constructor.

Reply:
Will do, just saw your comment above too.

Reply:
Done.